dsp-user/core/src/test/scala/dsp/user/domain/UserDomainSpec.scala

```diff
@@ -4,3 +4,161 @@
  */
 
 package dsp.user.domain
+
+import dsp.user.domain.User
+import dsp.user.domain._
+import dsp.valueobjects.User._
+import dsp.user.sharedtestdata.SharedTestData
+import zio.ZLayer
+import zio._
+import zio.test._
+
+/**
+ * This spec is used to test [[dsp.user.domain.UserDomain]].
+ */
+object UserDomainSpec extends ZIOSpecDefault {
+
+  def spec = (compareUsersTest + createUserTest + updateUserTest)
+
+  private val compareUsersTest = suite("compareUsers")(
+    test("compare two users") {
+      val user         = SharedTestData.user1
+      val userEqual    = SharedTestData.user1 // same as user, i.e. has same UserId
+      val userNotEqual = SharedTestData.user2
+
+      assertTrue(user.equals(userEqual)) &&
+      assertTrue(!user.equals(userNotEqual)) &&
+      assertTrue(user == userEqual) &&
+      assertTrue(user != userNotEqual)
+    }
+  )
+
+  private val createUserTest = suite("createUser")(
+    test("create a user") {
+      for {
+        id         <- SharedTestData.userId1.toZIO
```

Comment:
You can do the same thing without `.toZIO` (I think). Validations behave the same way like ZIOs.

Reply:
The `.toZIO` seems to be needed for the test to compile. But I now wrapped the whole for comprehension into a `.toZIO` instead of repeating it several times inside the for comprehension.